### PR TITLE
A Non-Semantic Extension to Programmatically Trigger Break Point

### PR DESCRIPTION
**Pull request by qingyuanzNV**
This extension introduces a non-semantic extension instruction DebugBreak as a hint for any attached shader debugger to hit a breakpoint programmatically. It is meant to provide similar feature like __debugbreak() in MSVC.

### CODE REVIEW

File: nonsemantic/NonSemantic.DebugBreak.asciidoc [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+2. Binary Form
+---------------
+
+[cols="1,1,2*3",width="100%"]
+|=====
+3+|[[DebugBreak]]*DebugBreak* +
+ +
+Hint the attached shader debugger to trigger a breakpoint and
+pause the execution. If no debugger is available, this instruction
+should either be ignored or terminte the program. +
```
Comment (r-potter) [inline nonsemantic/NonSemantic.DebugBreak.asciidoc:68]:
```suggestion
should either be ignored or terminate the program. +
```

File: nonsemantic/NonSemantic.DebugBreak.asciidoc [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+| Revision           | 1
+|========================================
+
+Dependencies
+------------
+
+This extension is written against the SPIR-V Specification,
+Version 1.6 Revision 2.
+
+This instruction set requires SPIR-V 1.0.
```
Comment (dgkoch) [inline nonsemantic/NonSemantic.DebugBreak.asciidoc:46]: Should this be 1.6 (which is the version that added non-semantic support to core)

**Comment by johnkslang** [comment]
Should it require the non-semantic instruction set extension instead of 1.6?

**Comment by dneto0** [comment]
> Should it require the non-semantic instruction set extension instead of 1.6?

I suspect this may be driven by how the tests are written. If they assume SPIR-V 1.6 then that's enough.
Also, it's a nice nudge into the future.

**Comment by dneto0** [comment]
> > Should it require the non-semantic instruction set extension instead of 1.6?
> 
> I suspect this may be driven by how the tests are written. If they assume SPIR-V 1.6 then that's enough. Also, it's a nice nudge into the future.

Oh, I didn't see the resolved conversation.  The reliance on 1.6 may be somewhat accidental. I'll step back now.

**Comment by qingyuanzNV** [comment]
I've updated the requirements to SPIR-V 1.0 and SPV_KHR_non_semantic_info.

**Pull request merged**